### ISSUE TITLE
Add resnet option for TF-SLIM to json

### ISSUE THREAD

**Issue by Russell91** [Russell91/TensorBox#115]
This should require downloading the resnet slim python file and the corresponding weights, editing utils/googlenet_load.py, and adding a json option for the user to choose this. Would be great to get some help if anyone wants to try.

**Comment by lukemurray** [comment]
Got anymore tips for this? I'm not having much luck.

**Comment by Russell91** [comment]
right, so I'd recommend you write something like print([(op.name, op.outputs[0].get_shape()) for op in slim.get_operations()]) where T['Mixed_5b'] is, and then look for a late layer that has the type of shape that you're looking for (e.g. 15x20).

**Comment by lukemurray** [comment]
And for `Mixed_3b` are we looking for a similar shape?

**Comment by Russell91** [comment]
Right, you got it. Mixed_3b should prefer to be earlier in the network. Basically that's after the rezoom, you go back and look at some lower level features.

**Comment by lukemurray** [comment]
Thanks, sorry I'm new to Tensorflow, struggling to output the operations... once I get that I should be able to to figure the rest out

**Comment by Russell91** [comment]
But... I would avoid using the rezoom network for the first run so you don't have to change two things.

**Comment by lukemurray** [comment]
Dumb question, how do I avoid using the rezoom network?

**Comment by lukemurray** [comment]
Ahh it is an option in the json :+1:

**Comment by Russell91** [comment]
Use the overfeat.json file, rather than overfeat_rezoom.json. Also, don't worry about being new, we get 150 visitors/day here and you're the only one whose tackling this challenge.

**Comment by lukemurray** [comment]
so `slim.get_operations()` doesn't exist. I can't seem to find a similar method in slim.

**Comment by Russell91** [comment]
If necessary, you can use [x for x in tf.get_operations() where 'Inception" in op.name] or something. I know I have done this before.

**Comment by lukemurray** [comment]
What is the difference between overfeat vs lstm?

**Comment by Russell91** [comment]
Lstm gives a major performance boost without rezoom, but unfortunately rezoom absorbs most of those gains. So I'd recommend you just use overfeat for now. It's a little more robust.

**Comment by lukemurray** [comment]
OK thanks. I might have to leave this to someone else or come back another time, I think it is over my head. Can't figure out how to print the layer names from `googlenet_load` and ignoring the `Mixed_3b` (rezoom) brings resize errors later on

**Comment by lukemurray** [comment]
What is the line `cnn = tf.reshape(cnn, [H['batch_size'] * H['grid_width'] * H['grid_height'], H['later_feat_channels']])` in train.py doing? It seems to be specific to inception_v1? as other models throw errors like `ValueError: Cannot reshape a tensor with 1685376 elements to shape (300, 832) (249600 elements)`

**Comment by Russell91** [comment]
Yeah, that hyper parameter was added specifically so that it could be
switched when changing the model. So you'll need to match it to the channel
dimension of the tensor you are outputting from resnet.

**Comment by lukemurray** [comment]
> So you'll need to match it to the channel
dimension of the tensor you are outputting from resnet
That is lost on me :)

**Comment by lukemurray** [comment]
OK, so I'm at this again. Actually trying to get Inception_v4 in but if I can understand how to do that, I'm sure I could get resnet in too.

Is there somewhere I can read more info on what/why you are doing certain things? Like
- You load the model, then reshape it (`[H['batch_size'] * H['grid_width'] * H['grid_height'], H['later_feat_channels']]`) What is the desired outcome for this?
- Same for the `lstm_input = lstm_input = tf.reshape(cnn * scale_down, (H['batch_size'] * grid_size, H['later_feat_channels']))`
- You require the `region_size` to be equal to the `image_height / grid_height` and `image_width / grid_width`. Again what does this achieve?
- Any ideas on this ` Assign requires shapes of both tensors to match. lhs shape= [8,12,128,768] rhs shape= [5,5,128,768]` happening intrain.py line 472 (`[x for x in tf.all_variables() if x.name.startswith...`
- where did you get the value for `later_feat_channels`?

Sorry, I'm not questioning the great work you have done, just trying to understand it so I can contribute something back. I've already got some OK results out of the box on a small dataset :)

**Comment by mdaniluk** [comment]
Did you manage to solve the issue with:

> Any ideas on this Assign requires shapes of both tensors to match. lhs shape= [8,12,128,768] rhs shape= [5,5,128,768] happening intrain.py line 472 ([x for x in tf.all_variables() if x.name.startswith...

**Comment by lukemurray** [comment]
No I haven't. I haven't had time to come back to this yet. 






On Sat, Dec 3, 2016 at 3:07 AM +1100, "Michal Daniluk" <notifications@github.com> wrote:












Did you manage to solve the issue with:



Any ideas on this Assign requires shapes of both tensors to match. lhs shape= [8,12,128,768] rhs shape= [5,5,128,768] happening intrain.py line 472 ([x for x in tf.all_variables() if x.name.startswith...
